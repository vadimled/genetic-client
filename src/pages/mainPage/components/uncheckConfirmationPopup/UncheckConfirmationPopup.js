import React from 'react';
import PropTypes from 'prop-types';
import { connect } from "react-redux";
import { Button } from 'antd';
import Portal from 'GenericComponents/portal';
import warningImg from 'Assets/warning-sign.svg';

import styles from './UncheckConfirmationPopup.module.scss';
// import { handleOnConfirmation, sendForConfirmation } from "Actions/confirmationActions";
// import { handleSelectedRow } from "Actions/tableActions";
// import { getUncheckConfirmationData } from "Store/selectors";

const UncheckConfirmationPopup = () => {
  return (
    <Portal>
      <div className={styles['uncheck-confirmation-popup']}>
        <header className="confirmation-header">
          <div className="header-title">
            <span
              className="warning-sign"
              style={{backgroundImage: `url(${warningImg})`}}
            />
            Uncheck Alert
          </div>
        </header>
        <div className="confirmation-content">
          Are you sure you want to erase the confirmation status?
        </div>
        <footer className="confirmation-footer">
          <div className="confirmation-btns">
            <Button
              className="confirmation-btn confirmation--link"
              onClick={() => false}
            >
              No
            </Button>
            <Button
              className="confirmation-btn"
              onClick={() => true}
            >
              Yes
            </Button>
          </div>
        </footer>
      </div>
    </Portal>
  );
};

UncheckConfirmationPopup.propTypes = {
  propName: PropTypes.string
};

UncheckConfirmationPopup.defaultProps = {
  propName: 'DefaultProp'
};


const mapDispatchToProps = () => {
  return {
    // handleOnConfirmation: data => dispatch(handleOnConfirmation(data)),
    // handleSelectedRow: data => dispatch(handleSelectedRow(data)),
    // sendForConfirmation: data => dispatch(sendForConfirmation(data))
  };
};

const mapStateToProps = () => {
  return {
    // data: getSelectedRows(state)
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(React.memo(UncheckConfirmationPopup));
