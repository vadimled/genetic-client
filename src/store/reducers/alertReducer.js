import createReducer from "./createReducer";
import actionsTypes from "../actionsTypes";

const initialState = {
  status: null, // error | warning | default,
  title: '',
  message: ''
};

const tableReducer = createReducer(initialState, {
  [actionsTypes.SET_ALERT]: (state, { payload }) => {
    if (!payload) payload = {};
    const { status = null, title = '', message = ''} = payload;

    return {
      status,
      title,
      message
    };
  }
});

export default tableReducer;
