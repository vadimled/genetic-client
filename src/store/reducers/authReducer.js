import _ from "lodash";
import actionsTypes from '../actionsTypes';
import createReducer from "./createReducer";

const initialState = {
  isLoggedIn: false,
  status: null,
  serverError: null,
  userEmail: null
};

const authReducer = createReducer(initialState, {
  [actionsTypes.LOGIN_SUCCESS]: (state, {payload}) => {

    return {
      ...state,
      isLoggedIn: true,
      ...payload
    };
  },

  [actionsTypes.LOGIN_RESULT]: (state, {payload}) => {
    return {
      ...state,
      isLoggedIn: !_.has(payload, "message"),
      serverError: null,
      ...payload
    };
  },

});


export default authReducer;
