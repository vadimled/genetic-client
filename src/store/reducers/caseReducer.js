import createReducer from "./createReducer";
import actionsTypes from "../actionsTypes";

const initialState = {
  gsid: null,
  panel_type: null,
  test_id: null,
  showTumorInfo: false,
  mutation_types: [],
  tumorInfo: {
    type: "",
    location: "",
    percent: -1
  }
};

const caseReducer = createReducer(initialState, {
  [actionsTypes.SET_CASE_DATA]: (state, { payload }) => {
    return {
      ...state,
      ...payload
    };
  },

  [actionsTypes.SET_TUMOR_INFO_MODE]: (state, { payload }) => {
    return {
      ...state,
      showTumorInfo: payload
    };
  },

  [actionsTypes.SET_TUMOR_INFO]: (state, { payload }) => {
    // const {type: [val](|| location || percent)} = payload;
    return {
      ...state,
      tumorInfo: {
        ...state.tumorInfo,
        ...payload
      }
    };
  }
});

export default caseReducer;
