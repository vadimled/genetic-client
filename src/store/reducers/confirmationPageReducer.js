import createReducer from "./createReducer";
import actionsTypes from "../actionsTypes";

const initialState = {
  metaData: null
};

const confirmationPageReducer = createReducer(initialState, {
  [actionsTypes.SET_CONFIRMATION_PAGE_METADATA]: (state, { payload }) => {
    return {
      ...state,
      metaData: payload
    };
  }
});

export default confirmationPageReducer;
