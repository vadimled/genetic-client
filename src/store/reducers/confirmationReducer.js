import createReducer from "./createReducer";
import actionsTypes from "../actionsTypes";
import { VALIDATION_FAILD_FIELDS } from "Utils/constants";

const initialState = {
  isOnConfirmation: false,
  data: []
};

const generateAdditionalConfirmationData = () => ({
  keyId: Math.random(),
  primer: "",
  fragmentSize: "",
  notes: "",
  validationFaildFields: []
});

const confirmationReducer = createReducer(initialState, {
  [actionsTypes.HANDLE_ON_CONFIRMATION]: (state, { payload }) => {
    return {
      ...state,
      isOnConfirmation: payload
    };
  },

  [actionsTypes.SET_CONFIRMATION_DATA]: (state, { payload }) => {
    let preparedData = [];
    if (payload) {
      preparedData = payload.map(row => {
        let newRow = {...row};
        if (!newRow.additionConfirmationData || !newRow.additionConfirmationData.length) {
          // initialize required filling out data
          newRow.additionConfirmationData = [generateAdditionalConfirmationData()];
        }
        return newRow;
      });
    }

    return {
      ...state,
      data: preparedData
    };
  },

  [actionsTypes.REMOVE_CONFIRMATION_ROW]: (state, { payload }) => {
    const filteredData = state.data.filter(row => row.id !== payload);

    return {
      ...state,
      data: filteredData
    };
  },

  [actionsTypes.HANDLE_CONFIRMATION_NOTES]: (state, { payload }) => {
    const { id, index, notes } = payload;
    const updatedData = state.data.map(row => {
      if (row.id === id) {
        row.additionConfirmationData[index].notes = notes;
      }
      return row;
    });

    return {
      ...state,
      data: updatedData
    };
  },

  [actionsTypes.HANDLE_CONFIRMATION_PRIMER]: (state, { payload }) => {
    const { id, index, value } = payload;
    const updatedData = state.data.map(row => {
      if (row.id === id) {
        let cRow = row.additionConfirmationData[index];
        cRow.primer = value;

        if (!cRow.primer || !/^[\d]*$/.test(cRow.primer)) {
          !cRow.validationFaildFields.includes(VALIDATION_FAILD_FIELDS.primer)
            && cRow.validationFaildFields.push(VALIDATION_FAILD_FIELDS.primer);
        }
        else {
          cRow.validationFaildFields = cRow.validationFaildFields
            .filter((key) => key !== VALIDATION_FAILD_FIELDS.primer);
        }
      }
      return row;
    });

    return {
      ...state,
      data: updatedData
    };
  },

  [actionsTypes.HANDLE_CONFIRMATION_FRAGMENT_SIZE]: (state, { payload }) => {
    const { id, index, value } = payload;
    const updatedData = state.data.map(row => {
      if (row.id === id) {
        let cRow = row.additionConfirmationData[index];
        cRow.fragmentSize = value;

        if (!cRow.fragmentSize || !/^[\d]*$/.test(cRow.fragmentSize)) {
          !cRow.validationFaildFields.includes(VALIDATION_FAILD_FIELDS.fragmentSize)
            && cRow.validationFaildFields.push(VALIDATION_FAILD_FIELDS.fragmentSize);
        }
        else {
          cRow.validationFaildFields = cRow.validationFaildFields
            .filter((key) => key !== VALIDATION_FAILD_FIELDS.fragmentSize);
        }
      }
      return row;
    });

    return {
      ...state,
      data: updatedData
    };
  },

  [actionsTypes.ADD_ADDITIONAL_CONFIRMATION_DATA]: (state, { payload }) => {
    const { id } = payload;
    const updatedData = state.data.map(row => {
      if (row.id === id) {
        row.additionConfirmationData.push(generateAdditionalConfirmationData());
      }
      return row;
    });

    return {
      ...state,
      data: updatedData
    };
  },

  [actionsTypes.REMOVE_ADDITIONAL_CONFIRMATION_DATA]: (state, { payload }) => {
    const { id, index } = payload;
    const updatedData = state.data.map(row => {
      if (row.id === id) {
        row.additionConfirmationData.splice(index, 1);
      }
      return row;
    });

    return {
      ...state,
      data: updatedData
    };
  },

});

export default confirmationReducer;


