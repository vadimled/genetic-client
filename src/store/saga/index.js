import {
  takeEvery,
  takeLatest
} from "redux-saga/effects";
import types from "Store/actionsTypes";

import {
  fetchTableData,
  fetchTestDataGenerator,
  fetchBAMFileGenerator,
  goToChrPositionIgvGenerator,
  sendForConfirmationGenerator,
  resultConfigLoadHgvsGenerator,
  resultConfigAddResultGenerator,
  resultConfigEditResultGenerator,
  fetchVariantDataGenerator,
  sendVariantClassGenerator,
  handleZygositySaga
} from "./sagas";

export function* watchSaga() {
  yield takeLatest(types.FETCH_TEST_DATA, fetchTestDataGenerator);
  yield takeLatest(types.FETCH_BAM_FILE, fetchBAMFileGenerator);
  yield takeLatest(types.GO_TO_CHR_POSITION_IGV, goToChrPositionIgvGenerator);
  yield takeLatest(types.SEND_FOR_CONFIRMATION, sendForConfirmationGenerator);
  yield takeEvery(types.FETCH_TABLE_DATA, fetchTableData);
  yield takeEvery(types.HANDLE_ZYGOSITY, handleZygositySaga);
  yield takeLatest(types.RESULT_CONFIG_LOAD_HGVS, resultConfigLoadHgvsGenerator);
  yield takeLatest(types.RESULT_CONFIG_ADD_RESULT, resultConfigAddResultGenerator);
  yield takeLatest(types.RESULT_CONFIG_EDIT_RESULT, resultConfigEditResultGenerator);
  yield takeLatest(types.FETCH_VARIANT_DATA, fetchVariantDataGenerator);
  yield takeLatest(types.SEND_VARIANT_CLASS, sendVariantClassGenerator);
}
