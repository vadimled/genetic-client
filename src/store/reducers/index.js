import { combineReducers } from "redux";
import filtersReducer from "./filtersReducer";
import tableReducer from "./tableReducer";
import variantsReducer from "./variantsReducer";
import igvReducer from "./igvReducer";
import confirmationReducer from "./confirmationReducer";
import alertReducer from "./alertReducer";
import resultConfigReducer from "./resultConfigReducer";
import variantPageReducer from "./variantPageReducer";
import testReducer from "./testReducer";
import testsPageReducer from "./testsPageReducer";
import evidenceConfigReducer from "./evidenceConfigReducer";
import confirmationPageReducer from "./confirmationPageReducer";
import coveragePageReducer from "./coveragePageReducer";
import finalReportReducer from "./finalReportReducer";

export default combineReducers({
  test: testReducer,
  tests: testsPageReducer,
  filters: filtersReducer,
  table: tableReducer,
  variants: variantsReducer,
  igv: igvReducer,
  confirmation: confirmationReducer,
  alert: alertReducer,
  resultConfig: resultConfigReducer,
  variantPage: combineReducers({
    pageData: variantPageReducer,
    evidenceConfig: evidenceConfigReducer
  }),
  confirmationPage: confirmationPageReducer,
  coveragePage: coveragePageReducer,
  finalReport: finalReportReducer
});

