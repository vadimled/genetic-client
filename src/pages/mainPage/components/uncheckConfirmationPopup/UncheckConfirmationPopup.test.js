import React from "react";
import { createStore, applyMiddleware } from "redux";
import createSagaMiddleware from 'redux-saga';
import { watchSaga } from "Store/saga";
import reducers from "Store/reducers";
import { renderWithRedux } from "Utils/test_helpers";
import { fireEvent } from "@testing-library/react";
import "jest-dom/extend-expect";
import UncheckConfirmationPopup from './UncheckConfirmationPopup';
import {
  fetchTableData,
  handleUncheckConfirmationData
} from "Actions/tableActions";
import {
  getUncheckConfirmationData
} from "Store/selectors";


const initSteps = () => {
  const sagaMiddleware = createSagaMiddleware();
  const { getByTestId, store, getAllByTestId, asFragment } = renderWithRedux(
    <UncheckConfirmationPopup />,
    createStore(reducers, applyMiddleware(sagaMiddleware))
  );
  sagaMiddleware.run(watchSaga);


  store.dispatch(fetchTableData());

  return {store, getByTestId, getAllByTestId, asFragment};
};

describe('UncheckConfirmationPopup', () => {

  it('UncheckConfirmationPopup snapshot', () => {
    const { asFragment } = initSteps();
    expect(asFragment()).toMatchSnapshot();
  });

  it('handle no btn', () => {
    const { getByTestId, store } = initSteps();

    // UncheckConfirmationPopup is only visible when uncheckConfirmationData2 has defined

    // find first item with status in table data and select one
    const tableData1 = store.getState().table.data;
    let itemId; // = tableData1[Object.keys(tableData1)[0]].id;
    for (let key in tableData1) {
      if (tableData1.hasOwnProperty(key)) {
        // if an item has already status we cannot select or unselect it to send for confirmation
        if (tableData1[key].status) {
          itemId = tableData1[key].id;
          break;
        }
      }
    }

    store.dispatch(handleUncheckConfirmationData({
      id: itemId,
      status: null
    }));

    const uncheckConfirmationData1 = getUncheckConfirmationData(store.getState());
    expect(uncheckConfirmationData1).toEqual({
      id: itemId,
      status: null
    });

    // and test no btn
    const noBtn = getByTestId('no-btn');

    fireEvent.click(noBtn);

    const uncheckConfirmationData2 = getUncheckConfirmationData(store.getState());
    expect(uncheckConfirmationData2).toEqual(null);
  });


  it('handle yes btn', () => {
    const { getByTestId, store } = initSteps();

    // UncheckConfirmationPopup is only visible when uncheckConfirmationData2 has defined

    // find first item with status in table data and select one
    const tableData1 = store.getState().table.data;
    let itemId; // = tableData1[Object.keys(tableData1)[0]].id;
    for (let key in tableData1) {
      if (tableData1.hasOwnProperty(key)) {
        // if an item has already status we cannot select or unselect it to send for confirmation
        if (tableData1[key].status) {
          itemId = tableData1[key].id;
          break;
        }
      }
    }

    store.dispatch(handleUncheckConfirmationData({
      id: itemId,
      status: null
    }));

    const uncheckConfirmationData1 = getUncheckConfirmationData(store.getState());
    expect(uncheckConfirmationData1).toEqual({
      id: itemId,
      status: null
    });

    // and test yes btn
    const yesBtn = getByTestId('yes-btn');

    fireEvent.click(yesBtn);

    const uncheckConfirmationData2 = getUncheckConfirmationData(store.getState());

    expect(uncheckConfirmationData2).toEqual(null);

    // status in row in tableData should be null
    const tableData2 = store.getState().table.data;
    const row2 = tableData2[itemId];
    expect(row2.status).toEqual(null);
  });
});
