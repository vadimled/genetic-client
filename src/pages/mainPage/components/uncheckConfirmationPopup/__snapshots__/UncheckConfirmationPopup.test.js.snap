// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UncheckConfirmationPopup UncheckConfirmationPopup snapshot 1`] = `<DocumentFragment />`;
